### PR TITLE
Devcontainer changes: minimised configs and added CI

### LINKED ISSUE #65: Test Docker image creation in CI
There are currently errors occurring in the Docker image creation. We need to catch these in CI.

### PR DESCRIPTION
**Pull request by axmmisaka**
This PR:

1. reduces the number of Codespaces configs from 3 to 1;
2. provides instructions in the `README.md` for switching between different versions of the Lingua Franca toolchain;
3. fixes #65; and
4. fixes the shell bugs and debugging issues which nondeterministically appear when starting up codespaces.

### CODE REVIEW

**Comment by axmmisaka** [comment]
Right now debugging is virtually impossible. I'm seeking help here: https://github.com/devcontainers/cli/issues/600

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    "hostRequirements": {
+      "cpus": 1
+    },
+    // Uncomment this line if GitHub displays no enough space
+    "waitFor": "postCreateCommand",
+    "updateContentCommand": "",
+    "onCreateCommand": "sudo mkdir -p /workspaces/.codespaces/.persistedshare/logs && sudo chown -R vscode /workspaces/.codespaces/.persistedshare/logs",
+    "postStartCommand": {
+      "setup-user-env": ["/bin/bash", "-c", "(bash -i ./utils/scripts/setup-user-env.bash >> /workspaces/.codespaces/.persistedshare/logs/stdout-env.log 2>>/workspaces/.codespaces/.persistedshare/logs/stderr-env.log) || (echo setup-user-env exited with nonzero code)"],
```
Comment (lhstrh) [inline .devcontainer/devcontainer.json:15]:
```suggestion
      "setup-user-env": ["/bin/bash", "-c", "(bash -i ./utils/scripts/setup-user-env.bash >> /workspaces/.codespaces/.persistedshare/logs/stdout-env.log 2>> /workspaces/.codespaces/.persistedshare/logs/stderr-env.log) || (echo setup-user-env exited with nonzero code)"],
```

File: .github/workflows/check-devcontainer.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          sed --in-place 's/\/\/ REMOVE-BY-CI //' .devcontainer/devcontainer.json
+          cat .devcontainer/devcontainer.json
+      - uses: devcontainers/ci@v0.3
+        with:
+          runCmd: lfc --version
+      - name: Show failure reason
+        if: failure()
+        run: |
+          echo "Failure reason:"
+          cat /home/runner/logs/*.log
```
Comment (lhstrh) [inline .github/workflows/check-devcontainer.yml:50]:
```suggestion
          cat /home/runner/logs/*.log

```

File: README.md [hunk 3]
```diff
@@ -14,11 +14,23 @@ A web-based VS Code editor, preloaded with the Lingua Franca extension, is acces
...
 
-#### :wrench: Alternate GitHub Codespaces configurations
-The default Codespaces configuration installs the nightly-built version of the Lingua Franca toolchain and checks out our examples repository. Links to alternate configurations are available below.
- * [Nightly build](https://github.com/codespaces/new?hide_repo_select=true&repo=569082724&ref=main&devcontainer_path=.devcontainer%2Fnightly-noexample%2Fdevcontainer.json)  
- * [Stable build](https://github.com/codespaces/new?hide_repo_select=true&repo=569082724&ref=main&devcontainer_path=.devcontainer%2Fstable-noexample%2Fdevcontainer.json) 
- * [Latest from `master`](https://github.com/codespaces/new?hide_repo_select=true&repo=569082724&ref=main&devcontainer_path=.devcontainer%2Fdev-noexample%2Fdevcontainer.json)  
+#### :wrench: Switch Lingua Franca Build Channels
+By default, the Lingua Franca runtime installed in the container will be the latest nightly version, freshly fetched when the container boots up. This fetching process is handled by a utility script called `setup-lf.bash`, located in the `utils/scripts` directory.
```
Comment (lhstrh) [inline README.md:18]:
```suggestion
By default, the Lingua Franca toolchain installed in the container will be the latest nightly version, freshly fetched when the container boots up. This fetching process is handled by a utility script called `setup-lf.bash`, located in the `utils/scripts` directory.
```

**Review by lhstrh — APPROVED** [review]
This looks great. I like the simplification of having a single `devcontainer.json`. I only have some minor suggestions for changes.

**Comment by axmmisaka** [comment]
Squashed these suggestions into relevant commits to ensure cleanliness.

**Pull request merged**